app/src/renderer/main.js

```diff
@@ -51,32 +54,37 @@ async function main () {
 
   let relayPort = getQueryParameter('relay_port')
   console.log('Expecting relay-server on port:', relayPort)
-
   console.log('Connecting to node:', nodeIP)
   const node = Node(nodeIP, relayPort)
 
   node.lcdConnected()
-  .then(connected => {
-    if (connected) {
-      axios.get(`http://localhost:${relayPort}/startsuccess`)
-    }
-  })
+    .then(connected => {
+      if (connected) {
+        axios.get(`http://localhost:${relayPort}/startsuccess`)
+      }
+    })
 
   const router = new Router({
     scrollBehavior: () => ({ y: 0 }),
     routes
   })
 
-  const store = Store({ node })
+  store = Store({ node })
 
-  let connected = await store.dispatch('checkConnection')
-  if (connected) {
-    store.dispatch('nodeSubscribe')
-    store.dispatch('showInitialScreen')
-    store.dispatch('subscribeToBlocks')
+  let error = getQueryParameter('error')
```

Comment:
i find it odd that we have to use url's even in a desktop app. anyone have any thoughts on this? not a big deal, just a curiosity.

Reply:
Currently this is the easiest solution. We can/should switch to a socket communication between the main thread and the browser.